src/wit_gen.rs

```diff
@@ -62,6 +62,21 @@ impl<F> TraceContext<F> {
 
         self.witness.step_instances.push(witness);
     }
+
+    // This function pads the rest of the circuit with the given StepTypeWGHandler
+    pub fn padding<Args, WG: Fn(&mut StepInstance<F>, Args) + 'static>(
+        &mut self,
+        step: &StepTypeWGHandler<F, Args, WG>,
+        args: Args,
+    ) {
+        let mut witness = StepInstance::new(step.uuid());
+
+        (*step.wg)(&mut witness, args);
+        let len = self.witness.step_instances.len();
+        for _ in 0..len {
```

Comment:
This loop from 0 to current length will duplicate the number of step instances. Not bring it to num_steps